sys var: add more contents to `tidb_remove_orderby_in_subquery` (#18530)
pingcap/docs: system-variables.md

- Type: Boolean
- Default value: Before v7.2.0, the default value is `OFF`. Starting from v7.2.0, the default value is `ON`.
- Specifies whether to remove `ORDER BY` clause in a subquery.

> - In the SQL standard, the behavior of `ORDER BY` is mainly used to sort the results of the top-level query. For subqueries, the standard does not require that the results be sorted by `ORDER BY`. 

Reviewer: ```suggestion
- In the ISO/IEC SQL standard, `ORDER BY` is mainly used to sort the results of top-level queries. For subqueries, the standard does not require that the results be sorted by `ORDER BY`. 
```

> - If subquery results are required to be sorted, it can usually be handled in the outer query, such as using the window function or using `ORDER BY` again in the outer query. Doing so ensures the order of the final result set.

Reviewer: ```suggestion
- To sort subquery results, you can usually handle it in the outer query, such as using the window function or using `ORDER BY` again in the outer query. Doing so ensures the order of the final result set.
```

> - If subquery results are required to be sorted, it can usually be handled in the outer query, such as using tje window function or using `ORDER BY` again in the outer query. Doing so ensures the order of the final result set.

Reviewer 2: tje window function?

> - If subquery results are required to be sorted, it can usually be handled in the outer query, such as using tje window function or using `ORDER BY` again in the outer query. Doing so ensures the order of the final result set.

Author: ```suggestion
- If subquery results are required to be sorted, it can usually be handled in the outer query, such as using the window function or using `ORDER BY` again in the outer query. Doing so ensures the order of the final result set.
```

> - In the SQL standard, the behavior of `ORDER BY` is mainly used to sort the results of the top-level query. For subqueries, the standard does not require that the results be sorted by `ORDER BY`. This behavior can be found in the SQL standard document, or its implementation details can be found in the database system documentation.
> - If subquery results are required to be sorted, it can usually be handled in the outer query, such as using the window function or using `ORDER BY` again in the outer query. Doing so ensures the order of the final result set.

Author: ```suggestion
- In the SQL standard, the behavior of `ORDER BY` is mainly used to sort the results of the top-level query. For subqueries, the standard does not require that the results be sorted by `ORDER BY`. 
- If subquery results are required to be sorted, it can usually be handled in the outer query, such as using the window function or using `ORDER BY` again in the outer query. Doing so ensures the order of the final result set.
```
